src/httprequest.rs

```diff
@@ -470,7 +472,9 @@ mod tests {
         let (router, _) = Router::new("/prefix/", routes);
         let info = router.default_route_info();
         assert!(info.has_route("/user/test.html"));
+        assert!(info.has_prefixed_route("/user/test.html"));
```

Comment:
In continuation from our discussion.
The current behavior makes method is confusing.

It matches both route with prefix or without prefix.

I think this behavior makes it confusing and it should match only in one fashion:
Match both prefix and path.
If prefix is missing from route, then match only against path.

If provides path missing prefix, then it cannot match route, this makes the function name `has_prefixed_route` corresponding to what it actually does.

It is not clear what is benefit matching path against route but ignoring prefix, if path doesn't contain it?